WPMedia/fusion-news-theme-blocks
# THEMES 1199 byline hover styles

**This PR goes with an `arc-themes-feature-pack` PR. You can find it [here](https://github.com/WPMedia/arc-themes-feature-pack/pull/424).**

## Description

The goal of this ticket is to update the hover style of the `share-bar-block` based on the [hover states audit](https://paper.dropbox.com/doc/Themes-hover-states-audit--B400R8DxsjiVy0YImyMBZ0B4Ag-JyHvxHuZttlIfj5BzjTI3). I made the following changes:
* `byline-block/themes/news.json`:
    * I gave the default selector styling so non-linked author names have the correct styling. A non-linked author's byline is now `#232b39` (`$text-color` in [this design guide]), and it has no text decoration
    * Linked authors have slightly different styling. They are also now colored `#232b39` at rest, but they are underlined
    * When linked authors are hovered over, the byline turns `#69778E` (`$text-color-subtle` in the design guide), and the text decoration disappears
* I updated the news styling in `./storybook/themes` to reflect the changes made to the block's themes directory
* As a result of these changes, some changes were also automatically made to the `site-styles` directory and `package-lock.json` in `arc-themes-feature-pack`
    * You can find a link to that PR [here](https://github.com/WPMedia/arc-themes-feature-pack/pull/424)
    * These are needed to run this code

## Jira Ticket

[THEMES-1199](https://arcpublishing.atlassian.net/browse/THEMES-1199)

## Acceptance Criteria

1. Rest: `$text-color` (`#232b39`) + underlined only when linked, when no link do not underline
2. Hover (when linked): `$text-color-subtle` (`#69778E`), no underline

## Test Steps

1. Checkout this branch: `git checkout THEMES-1199-byline-hover-styles`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/byline-block`
3. Open your local PageBuilder instance and open the 'All Blocks-Test-TA-Sandbox' page
    * The website type doesn't matter
    * Any other page with a byline block works too
4. Verify that the byline styling matches the desired values
    * **Note** that the author in the 'All Blocks-Test-TA-Sandbox' page is linked (if you click on his name it takes you to a separate page)

## Effect Of Changes

### Before

https://github.com/WPMedia/arc-themes-blocks/assets/103296586/b523abe2-2219-403f-9f2f-cc9ccbc54865

### After

https://github.com/WPMedia/arc-themes-blocks/assets/103296586/8069ad01-8540-4483-aa72-90e83f2422c3

***Note** that Tariq is a linked author*

## Dependencies or Side Effects

This PR is dependent on some changes to `arc-themes-feature-pack`. The PR is [here](https://github.com/WPMedia/arc-themes-feature-pack/pull/424).

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-1199]: https://arcpublishing.atlassian.net/browse/THEMES-1199?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
